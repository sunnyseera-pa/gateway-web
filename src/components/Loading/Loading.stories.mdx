import { Meta, Story, Canvas } from '@storybook/addon-docs';
import Loading from './Loading';

<Meta
    title='Core/Loading'
    component={Loading}
    args={{
        text: 'loading...',
        subText: 'still loading...',
        timeout: 2000,
    }}
/>

<Canvas>
    <Story name='Basic'>{props => <Loading {...props} />}</Story>
</Canvas>
