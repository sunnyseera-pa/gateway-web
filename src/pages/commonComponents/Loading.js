import React, { useEffect, useState } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Image from 'react-bootstrap/Image';
import PropTypes from 'prop-types';

const Loading = props => {
    const { text, subText, timeout } = { ...props };
    const [showSubText, setShowSubText] = useState(false);
    useEffect(() => {
        setTimeout(() => {
            setShowSubText(true);
        }, timeout);
    });
    return (
        <Row className='mt-2' {...props}>
            <Col>
                <div className='rectangle'>
                    <div className='gray800-14' style={{ textAlign: 'center' }}>
                        <Image src={require('../../images/Loader.gif')} />
                    </div>
                    <div className='gray800-14' style={{ textAlign: 'center' }}>
                        <p>{text}</p>
                        <p>{showSubText ? subText : ''}</p>
                    </div>
                </div>
            </Col>
        </Row>
    );
};

Loading.propTypes = {
    text: PropTypes.string,
    subText: PropTypes.string,
    timeout: PropTypes.number,
};

Loading.defaultProps = {
    text: 'Loading...',
    subText: '...sorry, this is taking longer than usual',
    timeout: 5000,
};

export default Loading;
