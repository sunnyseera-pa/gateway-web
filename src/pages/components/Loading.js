import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Image from 'react-bootstrap/Image';


class Loading extends React.Component {

    render() {
        return (
            <Row className="mt-2">
                <Col>
                    <div className="Rectangle">
                        <div className="Gray800-14px" style={{ textAlign: 'center' }}>
                            <Image src={require("../../images/Loader.gif")} />
                        </div>
                        <div className="Gray800-14px" style={{ textAlign: 'center' }}>
                            Loading...
                        </div>
                    </div>
                </Col>
            </Row>
        )
    }
}

export default Loading;