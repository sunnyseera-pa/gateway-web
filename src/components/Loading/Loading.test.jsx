import React from 'react';
import { render } from '@testing-library/react';
import Loading from './Loading';

let wrapper;
jest.setTimeout(30000);
const props = {
    text: 'loading...',
    subText: 'still loading....',
    timeout: 3000,
};

describe('Given the Loading component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            wrapper = render(<Loading {...props} />);
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Then contains the correct text', () => {
            expect(wrapper.getByText('loading...')).toBeTruthy();
        });

        it('Then contains the sub text', async () => {
            await new Promise(r => setTimeout(r, 4000));
            expect(wrapper.getByText('still loading....')).toBeTruthy();
        });
    });
});
