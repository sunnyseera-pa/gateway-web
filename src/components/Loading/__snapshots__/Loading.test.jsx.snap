// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Loading component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="mt-2 row"
  >
    <div
      class="col"
    >
      <div
        class="rectangle"
      >
        <div
          class="gray800-14"
          style="text-align: center;"
        >
          <img
            class=""
            src=""
          />
        </div>
        <div
          class="gray800-14"
          style="text-align: center;"
        >
          <p>
            loading...
          </p>
          <p />
        </div>
      </div>
    </div>
  </div>
</div>
`;
